src/coord/src/catalog/builtin.rs

```diff
@@ -361,11 +353,64 @@ lazy_static! {
             .with_column("global_id", ScalarType::String.nullable(false))
             .with_column("schema_id", ScalarType::Int64.nullable(false))
             .with_column("views", ScalarType::String.nullable(false)),
-        id: GlobalId::System(2025),
-        index_id: GlobalId::System(2026),
+        id: GlobalId::System(2023),
+        index_id: GlobalId::System(2024),
     };
 }
 
+pub const MZ_RELATIONS: BuiltinView = BuiltinView {
+    name: "mz_relations",
+    schema: MZ_CATALOG_SCHEMA,
+    sql: "CREATE VIEW mz_relations (global_id, oid, schema_id, name, type) AS
+      SELECT global_id, oid, schema_id, tables, 'table' FROM mz_catalog.mz_tables
+UNION SELECT global_id, oid, schema_id, sources, 'source' FROM mz_catalog.mz_sources
+UNION SELECT global_id, oid, schema_id, views, 'view' FROM mz_catalog.mz_views",
+    id: GlobalId::System(3000),
+    needs_logs: false,
+};
+
+pub const MZ_OBJECTS: BuiltinView = BuiltinView {
+    name: "mz_objects",
+    schema: MZ_CATALOG_SCHEMA,
+    sql: "CREATE VIEW mz_objects (global_id, oid, schema_id, name, type) AS
+    SELECT global_id, oid, schema_id, name, type FROM mz_catalog.mz_relations
+UNION
+    SELECT global_id, oid, schema_id, sinks, 'sink' FROM mz_catalog.mz_sinks
+UNION
+    SELECT mz_indexes.global_id, mz_indexes.oid, schema_id, indexes, 'index'
+    FROM mz_catalog.mz_indexes
+    JOIN mz_catalog.mz_relations ON mz_indexes.on_global_id = mz_relations.global_id",
+    id: GlobalId::System(3001),
+    needs_logs: false,
+};
+
+// For historical reasons, this view does not properly escape identifiers. For
+// example, a table named 'cAp.S' in the default database and schema will be
+// rendered as `materialize.public.cAp.S`. This is *not* a valid SQL identifier,
+// as it has an extra dot, and the capitals will be folded to lowercase. This
+// view is thus only fit for use in debugging queries where the names are for
+// human consumption. Applications should instead pull the names of individual
+// components out of mz_objects, mz_schemas, and mz_databases to avoid these
+// escaping issues.
+//
+// TODO(benesch): deprecate this view.
```

Comment:
What do you want to replace this with in the future?

Reply:
Ideally nothing! The names of things are in `mz_objects` already, so most queries can just be written to use the name from there. If you do need the fully qualified name, joining against `mz_schemas` and `mz_databases` means you'll get the database, schema, and item name in three separate columns, which is much more reliable.